ledger/internal/eval.go

```diff
@@ -41,13 +41,17 @@ import (
 type LedgerForCowBase interface {
 	BlockHdr(basics.Round) (bookkeeping.BlockHeader, error)
 	CheckDup(config.ConsensusParams, basics.Round, basics.Round, basics.Round, transactions.Txid, ledgercore.Txlease) error
-	LookupWithoutRewards(basics.Round, basics.Address) (basics.AccountData, basics.Round, error)
+	LookupWithoutRewards(basics.Round, basics.Address) (ledgercore.AccountData, basics.Round, error)
+	LookupResource(basics.Round, basics.Address, basics.CreatableIndex, basics.CreatableType) (ledgercore.AccountResource, error)
 	GetCreatorForRound(basics.Round, basics.CreatableIndex, basics.CreatableType) (basics.Address, bool, error)
 }
 
 // ErrRoundZero is self-explanatory
 var ErrRoundZero = errors.New("cannot start evaluator for round 0")
 
+// ErrNotInCowCache is returned when a lookup method requests a cached value, but it can't be found
+var ErrNotInCowCache = errors.New("can't find object in roundCowBase")
```

Comment:
Isn't the "roundCowBase" the parent interface? So that's the place that wasn't looked into at all, if only the cache was examined.

Reply:
`roundCowBase` is the last level being tested before hitting the actual ledger. and the code seems to be correct here. i.e. I'm looking at roundCowBase.lookupAssetHolding; maybe it's not the case in other places ?
I'm not opposing to change this into `can't find object in cow cache`, if that make it any better though.

Reply:
I think it makes the error message make sense, even if propagated up through roundCowState lookups. For example:
```
func (cb *roundCowState) lookupAssetParams(addr basics.Address, aidx basics.AssetIndex, fromCache bool) (ledgercore.AssetParamsDelta, bool, error) {
	params, ok := cb.mods.Accts.GetAssetParams(addr, aidx)
	if ok {
		return params, ok, nil
	}

	return cb.lookupParent.lookupAssetParams(addr, aidx, fromCache)
}
```
As written, this function says it was unable to find the asset in "roundCowBase" but the reality is that it was not able to find it in the roundCowState OR the roundCowBase.

Reply:
oh - I see what you mean. You're saying that the error itself is good as long it is was limited to `roundCowBase`. however, once it's propagated upstream, the scope of that error is getting wider, and it won't align with the original error message anymore.

Given that the gains of detecting the error message, and modifying it is pretty much non-existing, I'll follow your suggestion and make this change with some extra documentation for that use case.

Reply:
yes I guess roundCowBase is the end of the line just before it hits the DB, but a cached result could have been found in a cache in one of the "mods" caches in roundCowState or the cache in roundCowBase.. updating..